pkg/decoder/line_buffer.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package decoder
+
+import (
+	"bytes"
+)
+
+// truncated is the warning we add at the beginning or at the end of a truncated message
+var truncated = []byte("...TRUNCATED...")
+
+// LineBuffer accumulates lines in buffer to form and send outputs to outputChan
+type LineBuffer struct {
+	outputChan chan *Output
+	buffer     *bytes.Buffer
+	contentLen int
+}
+
+// NewLineBuffer returns a new LineBuffer
+func NewLineBuffer(outputChan chan *Output) *LineBuffer {
+	buffer := bytes.Buffer{}
+	return &LineBuffer{
+		outputChan: outputChan,
+		buffer:     &buffer,
+	}
+}
+
+// IsEmpty returns true if buffer does not contain any line
+func (l *LineBuffer) IsEmpty() bool {
+	return l.contentLen == 0
+}
+
+// Length returns the length of buffer which might be different from contentLen because of escaped '\n'
+func (l *LineBuffer) Length() int {
+	return l.buffer.Len()
+}
+
+// Add stores line in buffer
+func (l *LineBuffer) Add(line *Line) {
+	l.buffer.Write(line.content)
+	l.contentLen += len(line.content) + 1 // add 1 for '\n'
```

Comment:
`// add 1 for '\n'` is not super clear. Could we write a paragraph explaining exactly why we're doing that?

Reply:
I added a better explanation in the header of the struct definition, WDYT ?

Reply:
ok